packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -1124,7 +1124,8 @@ class FlutterView extends ServiceObject {
       // launch errors.
       if (event.kind == ServiceEvent.kIsolateRunnable) {
         printTrace('Isolate is runnable.');
-        completer.complete(null);
+        if (!completer.isCompleted)
+          completer.complete(null);
```

Comment:
Looks like arg for `complete()` is optional. We could omit the `null` here, or specify it for clarity.